src/System.Runtime.Serialization.Xml/tests/Performance/XmlUTF8TextReaderTests.cs

```diff
@@ -0,0 +1,141 @@
+
+        // <Root>?????(invalid)EOF
+        private static byte[] invalidExtraEFBFBF_EOF = new byte[] { 0x3C, 0x52, 0x6F, 0x6F, 0x74, 0x3E, 0xEF, 0x80, 0x80, 0xEF, 0x80, 0x80, 0xEF, 0x80, 0x80, 0xEF, 0x80, 0x80, 0xEF, 0x80, 0x80, 0xEF, 0xBF, 0xBF };
+        private static byte[] invalidExtraEFBF_EOF = new byte[] { 0x3C, 0x52, 0x6F, 0x6F, 0x74, 0x3E, 0xEF, 0x80, 0x80, 0xEF, 0xBF };
+        private static byte[] invalidExtraEF_EOF = new byte[] { 0x3C, 0x52, 0x6F, 0x6F, 0x74, 0x3E, 0xEF, 0x80, 0x80, 0xEF };
+
+        // Dictionary<byteArrayToTest, shouldPass>
+        private static Dictionary<byte[], bool> tests = new Dictionary<byte[], bool>();
+
+        private static byte[] GenerateTestAsciiBytes(int numChars)
+        {
+            byte[] byteArray = new byte[numChars + 13];
+
+            int offset = 0;
+
+            byteArray[offset++] = 0x3C;     // < 
+            byteArray[offset++] = 0x52;     // R
+            byteArray[offset++] = 0x6F;     // o
+            byteArray[offset++] = 0x6F;     // o
+            byteArray[offset++] = 0x74;     // t
+            byteArray[offset++] = 0x3E;     // >
+
+            for (int i = 0; i < numChars; i++)
+            {
+                byteArray[offset++] = (byte)((int)'a' + i % 10);
+            }
+
+            byteArray[offset++] = 0x3C;     // < 
+            byteArray[offset++] = 0x2F;     // /
+            byteArray[offset++] = 0x52;     // R
+            byteArray[offset++] = 0x6F;     // o
+            byteArray[offset++] = 0x6F;     // o
+            byteArray[offset++] = 0x74;     // t
+            byteArray[offset++] = 0x3E;     // >
+
+            return byteArray;
+        }
+
+        private static byte[] GenerateTestEFBytes(int numChars)
+        {
+            byte[] byteArray = new byte[3 * numChars + 13];
+
+            int offset = 0;
+
+            byteArray[offset++] = 0x3C;     // < 
+            byteArray[offset++] = 0x52;     // R
+            byteArray[offset++] = 0x6F;     // o
+            byteArray[offset++] = 0x6F;     // o
+            byteArray[offset++] = 0x74;     // t
+            byteArray[offset++] = 0x3E;     // >
+
+            for (int i = 0; i < numChars; i++)
+            {
+                byteArray[offset++] = 0xEF;
+                byteArray[offset++] = 0xBF;
+                byteArray[offset++] = (byte)(0x0A + numChars % 10);
+            }
+
+            byteArray[offset++] = 0x3C;     // < 
+            byteArray[offset++] = 0x2F;     // /
+            byteArray[offset++] = 0x52;     // R
+            byteArray[offset++] = 0x6F;     // o
+            byteArray[offset++] = 0x6F;     // o
+            byteArray[offset++] = 0x74;     // t
+            byteArray[offset++] = 0x3E;     // >
+
+            return byteArray;
+        }
+
+        const int FiveMega = 5 * 1024 * 1024;
+
+        public void RunTest(byte[] testBytes, int iterations)
```

Comment:
The method name 'RunTest' seems unspecific, considering it's in the class named XmlUTF8TextReaderTests. Can it be 'RunPerfTest' or something more meaningful?